src/ar_kernel.cpp

```diff
@@ -290,7 +259,8 @@ bool ar_kernel_increment_tick_count(unsigned ticks)
             ar_list_node_t * next = node->m_next;
 
             // Is it time to wake this thread?
-            if (g_ar.tickCount >= thread->m_wakeupTime)
+            //TODO: would it be safe to cache tick_count?
```

Comment:
I was wondering about that too. It's probably safe to cache it within a single function like this one, or `ar_kernel_scheduler()`.